### PR TITLE
Feat/UI polish plus navigation

### PR DESCRIPTION
**Pull request by wderocco8**
## What does this PR do?

Added some notable UI polish to the frontend:
- glowing effect around cards on hover (or hold on mobile)
- adjusted `NavSidebar` to use flowbite components
- fixed layout issues with navbar
- implemented `dark mode` across the whole site (persistent on page refresh)

## Type of change

- [ ] Fix: Bug fix (non-breaking change which fixes an issue)
- [x] Refactor: Any code refactoring
- [ ] Chore: technical debt, workflow improvements
- [x] Feature: New feature (non-breaking change which adds functionality)
- [ ] Documentation: This change requires a documentation update

### CODE REVIEW

File: src/app/attendance/page.tsx
Comment (jinyoungbang) [inline src/app/attendance/page.tsx:1]: I think it'd be great if we could have an animation, but let's have a separate PR for that later!

File: src/app/checkin/[id]/page.tsx [hunk 1]
```diff
@@ -14,6 +11,7 @@ import {
   AlertDescription,
   AlertTitle,
 } from "@/components/ui/alert"
+import { Badge, TextInput, Button } from "flowbite-react"
```
Comment (jinyoungbang) [inline src/app/checkin/[id]/page.tsx:14]: Was there a reason why we switched back to flowbite? When I designed this website in mind, I wanted to go for more of a minimal look and tried keeping styles consistent as possible using the UI library that was being used (https://ui.shadcn.com/)

File: src/app/checkin/success/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// import '../../globals.css'
+import UserSessionProvider from '@/app/providers/userSessionProvider'
+import NavSidebar from '@/components/NavSidebar'
+import type { Metadata } from 'next'
+
+
+export const metadata: Metadata = {
+  title: 'WhyPhi - Rush Checkin Error',
```
Comment (jinyoungbang) [inline src/app/checkin/success/layout.tsx]: File says success but title says error.

File: src/app/page.tsx [hunk 3]
```diff
@@ -23,24 +35,24 @@ export default function Home() {
           <div className="max-w-xl lg:max-w-3xl">
             <a className="block text-blue-600" href="https://bupct.com/">
               <img
-                src="/pct-logo.png"
+                src={mode === 'dark' ? '/pct-logo-dark.png' : '/pct-logo.png'}
```
Comment (jinyoungbang) [inline src/app/page.tsx:38]: nice touch!

**Review by jinyoungbang — APPROVED** [review]
See comments, curious about your thoughts - impressive work though!

File: src/app/checkin/[id]/page.tsx (hunk 1 above)
Comment (wderocco8) [reply src/app/checkin/[id]/page.tsx:14]: Mainly for consistency with Dark Mode support, I tried to only switch it for a few things, and am good with switching back if you feel strongly. But I kept the styles very similar to the custom components you were using, and think that Flowbite actually makes it easier to have a more consistent look and feel.

Lmk what you think!

**Pull request merged**